python/src/aac/lang/language_context.py

```diff
@@ -0,0 +1,182 @@
+        """
+
+        extension_definitions = get_definitions_by_root_key("ext", definitions)
+
+        # Avoiding the use of sorted() since we already deepcopy each definition as we add it to the context.
+        for definition in definitions:
+            if definition not in extension_definitions:
+                self.add_definition_to_context(definition)
+
+        for extension_definitions in extension_definitions:
+            self.add_definition_to_context(extension_definitions)
+
+    def get_root_keys(self) -> list[str]:
+        """
+        Get the list of root keys as defined in the LanguageContext.
+
+        Returns a list of strings indicating the current root keys in the active context. These
+        keys may differ from those provided by the core spec since the LanguageContext applies definitions
+        from active plugins and user files, which may extend the set of root keys.
+        See :py:func:`aac.spec.get_root_keys()` for the list of root keys provided by the unaltered core AaC DSL.
+
+        Returns:
+            A list of strings, one entry for each root key in the LanguageContext.
+        """
+        return [field.get("name") for field in self.get_root_fields()]
+
+    def get_root_fields(self) -> list[dict]:
+        """
+        Get the list of root fields as defined in the LanguageContext.
+
+        Returns a list of dictionaries that consist of the root fields in the active context. These
+        keys may differ from those provided by the core spec since the LanguageContext applies definitions
+        from active plugins and user files, which may extend the set of root keys.
+
+        Returns:
+            A list of dictionaries, one dictionary for each root field including name and type.
+        """
+        root_definition = self.get_definition_by_name("root")
+
+        if root_definition:
+            return search_definition(root_definition, ["data", "fields"])
+        else:
+            return []
+
+    def get_primitives_definition(self) -> list[str]:
+        """
+        Return the primitive type definition in the LanguageContext.
+
+        Returns:
+            The definition that defines the primitive types.
+        """
+        return self.get_definition_by_name("Primitives")
+
+    def get_primitive_types(self) -> list[str]:
+        """
+        Get the list of primitive types as defined in the LanguageContext.
+
+        Returns a list of strings indicating the currently defined primitive types in the active context.
+        These types may differ from those provided by the core spec since the LanguageContext applies definitions
+        from active plugins and user files, which may extend the set of root keys.
+        See :py:func:`aac.spec.get_primitives()` for the list of root keys provided by the unaltered core AaC DSL.
+
+        Returns:
+            A list of strings, one entry for each root name available in the LanguageContext.
```

Comment:
```suggestion
        Returns a list of strings indicating the currently defined primitive types in the active context.
        
        Returns:
            A list of strings, one entry for each primitive type defined in the LanguageContext.
            
            These types may differ from those provided by the core spec since the LanguageContext applies definitions
            from active plugins and user files, which may extend the set of root keys.
            See :py:func:`aac.spec.get_primitives()` for the list of root keys provided by the unaltered core AaC DSL.
```